lib/exceptions.h

```diff
@@ -29,6 +30,11 @@ constexpr char ANSI_RED[]  = "\e[31m";
 constexpr char ANSI_BLUE[] = "\e[34m";
 constexpr char ANSI_CLR[]  = "\e[0m";
 
+/// Checks if stderr is redirected to a file
+inline bool is_cerr_redirected() {
```

Comment:
I would suggest having a static bool to save the result after the first call. It won't change later, so you can avoid calling ttyname. It is unlikely that error is called many times, but it is still healthier to save the value.